### PR TITLE
500 cigarettes, and one edible component update

### PR DESCRIPTION
**Pull request by Maia-J**
## About The Pull Request

Makes cigarettes edible, and adds an achievement for eating 500 of them.

The examine text also hides all the interesting info about being able to eat it
image
image

This means the edible component has been touched up with a variable to hide examines.

## Why It's Good For The Game

An interesting challenge for achievement hunters. Also it's kinda funny and goes in line with the Sisyphus one

## Changelog

:cl:
add: Achievement for eating 500 cigarettes
balance: Cigarettes are now edible
code: Adds a variable to hide the food examine on the edible element
/:cl:

### CODE REVIEW

File: code/game/objects/items/cigs_lighters.dm [hunk 1]
```diff
@@ -194,12 +197,41 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+		return
+	var/ckey = eater.client.ckey
+	// We must have more!
+	GLOB.cigarette_eaters[ckey]++
+	if(GLOB.cigarette_eaters[ckey] >= 500)
+		eater.client.give_award(/datum/award/achievement/misc/cigarettes)
+
+/obj/item/cigarette/examine(mob/user)
+	. = ..()
+	if(!is_species(user, /datum/species/human))
```
Comment (Jacquerel) [inline code/game/objects/items/cigs_lighters.dm:232]: Why?

Comment (Majkl-J) [reply code/game/objects/items/cigs_lighters.dm:232]: Simply because there's no helper for it (Also I don't think a human would wish to taste it)

**Comment by Jolly-66** [comment]
300px-Gentlemen

**Comment by TheVekter** [comment]
That joke sucked, and of all the actually kinda good stuff in Orville, it makes me mad this is the one that took off.

**Comment by Ghommie** [comment]
> image

I think you need to find a way to hide these edible lines from cigarettes. To quote it: "What are you doing? It is not food."

File: code/game/objects/items/cigs_lighters.dm (hunk 1 above)
Comment (Ghommie) [reply code/game/objects/items/cigs_lighters.dm:232]: Add another check for the TRAIT_NOHUNGER trait. That's all.

**Comment by Majkl-J** [comment]
> I think you need to find a way to hide these edible lines from cigarettes. To quote it: "What are you doing? It is not food." Also, yey, moths can eat cigs again.

Time to do some edible component refactoring B)

**Comment by Metekillot** [comment]
I been grinding em for the NIC juice already. Honestly, make ingestion of nicotine through (ignores_stomach = FALSE) cause horrible nausea without the strong stomach perk

**Comment by Exester509** [comment]
> That joke sucked, and of all the actually kinda good stuff in Orville, it makes me mad this is the one that took off.

Agreed. This will get old in a month or two anyways

File: code/game/objects/items/cigs_lighters.dm (hunk 1 above)
Comment (MrMelbert) [inline code/game/objects/items/cigs_lighters.dm:232]:
```suggestion
	if(!ishumanbasic(user))
```

File: code/game/objects/items/cigs_lighters.dm [hunk 2]
```diff
@@ -194,12 +197,41 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+		. += span_boldwarning("You are curious to taste it")
```
Comment (MrMelbert) [inline code/game/objects/items/cigs_lighters.dm:233]:
```suggestion
		. += span_boldwarning("You are curious to taste it...")
```

File: code/game/objects/items/cigs_lighters.dm [hunk 3]
```diff
@@ -129,6 +129,9 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 //////////////////
 //FINE SMOKABLES//
 //////////////////
+
+GLOBAL_LIST_EMPTY(cigarette_eaters) // Need to keep track of up to 500 cigarettes
```
Comment (MrMelbert) [inline code/game/objects/items/cigs_lighters.dm:133]: Could just make this a static list on `/obj/item/cigarette`

File: code/game/objects/items/cigs_lighters.dm [hunk 4]
```diff
@@ -194,12 +197,41 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
 
+	// "It is called a cigarette"
+	AddComponent(/datum/component/edible,\
+		initial_reagents = list_reagents,\
+		food_flags = null,\
+		foodtypes = JUNKFOOD,\
+		volume = 50,\
+		eat_time = 0 SECONDS,\
+		tastes = list("A never before experienced flavour.", "Finally sitting down after standing your entire life"),\
+		eatverbs = list("Chomps"),\
```
Comment (MrMelbert) [inline code/game/objects/items/cigs_lighters.dm:208]:
```suggestion
		eatverbs = list("chomps"),\
```

File: code/game/objects/items/cigs_lighters.dm [hunk 5]
```diff
@@ -194,12 +197,41 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
 	inhand_icon_state = inhand_icon_off
 
+	// "It is called a cigarette"
+	AddComponent(/datum/component/edible,\
+		initial_reagents = list_reagents,\
+		food_flags = null,\
+		foodtypes = JUNKFOOD,\
+		volume = 50,\
+		eat_time = 0 SECONDS,\
+		tastes = list("A never before experienced flavour.", "Finally sitting down after standing your entire life"),\
```
Comment (MrMelbert) [inline code/game/objects/items/cigs_lighters.dm:207]:
```suggestion
		tastes = list("a never before experienced flavour.", "finally sitting down after standing your entire life"),\
```

File: code/game/objects/items/cigs_lighters.dm (hunk 1 above)
Comment (Majkl-J) [reply code/game/objects/items/cigs_lighters.dm]: On it sir

File: code/game/objects/items/cigs_lighters.dm [hunk 6]
```diff
@@ -194,12 +197,41 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+	var/ckey = eater.client.ckey
+	// We must have more!
+	GLOB.cigarette_eaters[ckey]++
+	if(GLOB.cigarette_eaters[ckey] >= 500)
+		eater.client.give_award(/datum/award/achievement/misc/cigarettes)
+
+/obj/item/cigarette/examine(mob/user)
+	. = ..()
+	if(!ishumanbasic(user))
+		. += span_boldwarning("You are curious to taste it...")
```
Comment (nikothedude) [inline code/game/objects/items/cigs_lighters.dm:233]: once the meme dies out people are just gonna get really confused by this, idk this just feels out of place

File: code/datums/achievements/misc_achievements.dm [hunk 7]
```diff
@@ -232,3 +232,9 @@
 	desc = "Successfully carry a boulder from Lavaland all the way to Centcom, without ever dropping it. We must imagine you're happy to unlock this."
 	database_id = MEDAL_SISYPHUS
 	icon_state = "sisyphus"
+
+/datum/award/achievement/misc/cigarettes
+	name = "Unhealthy snacks"
+	desc = "You were curious to taste it. And then another. You must have more!"
```
Comment (Iamgoofball) [inline code/datums/achievements/misc_achievements.dm:238]: Do we really need to immortalize fucking Seth McFarlane writing in our codebase?

**Comment by Iamgoofball** [comment]
Do we really need to immortalize fucking Seth MacFarlane writing in our codebase?

File: code/datums/achievements/misc_achievements.dm (hunk 7 above)
Comment (Majkl-J) [reply code/datums/achievements/misc_achievements.dm:238]: yes

**Comment by Ghommie** [comment]
I see no edible component update.

**Comment by Majkl-J** [comment]
> I see no edible component update.

I forgot to push and almost went to bed. Apologies

**Comment by TheVekter** [comment]
I, too, enjoy rolling my dumbass references in with actual improvements to the game so they'll get merged.

Whatever, I respect the hustle.

File: code/datums/components/food/edible.dm [hunk 8]
```diff
@@ -28,6 +28,8 @@ Behavior that's still missing from this component that original food items had t
 	var/junkiness = 0
 	///Message to send when eating
 	var/list/eatverbs
+	///Callback to be ran to see if the food can be eaten. Specifically used for carbon handling.
+	var/datum/callback/check_edible
```
Comment (Ghommie) [inline code/datums/components/food/edible.dm:32]: This is currently unused, any plan for it?

**Comment by SomeoneYouProbablyKnow** [comment]
Kinda weird. You gonna have a lot of people accidentally eating them because all you have to do is click on yourself (or someone else with it). There isn't even a special action for it.

**Comment by Majkl-J** [comment]
> I, too, enjoy rolling my dumbass references in with actual improvements to the game so they'll get merged.
> 
> Whatever, I respect the hustle.

Admittedly the first version came without the improvements (And I'd do them anyhow even if this was closed)

File: code/datums/components/food/edible.dm (hunk 8 above)
Comment (Majkl-J) [reply code/datums/components/food/edible.dm:32]: Yup, I do have some ideas with some specific antag/species only edibles, but I can also just remove it and add it with the PR implementing said ideas. Just felt weird that it allows for different custom effects but not for actually custom checking if you can eat it and whatnot

Comment (Ghommie) [reply code/datums/components/food/edible.dm:32]: If it's something for the future, then don't include this change in this PR. It's pretty small and doesn't affect this feature in any way..

Comment (Majkl-J) [reply code/datums/components/food/edible.dm]: Alright, fixed up

**Review by Ghommie — CHANGES_REQUESTED** [review]
I've a couple nits with the cheevo icon. I appreciate the shape and the colors are fine, but it lacks a few inlines that would otherwise give the sensation that it's a huge pile of cigs rather than a single object. Also I find the blue light reflected on the cigs filters out of place. I understand you were mainly inspired from that shot with the pile of cigarettes forming from sci-fi aether, but unlike that scene, there's no light source that'd make sense of it in this cheevo.

Sorry for not having said it earlier. I'd have liked to take care of it myself, but it turns out I'm a bit busy, so if you would...

**Comment by Majkl-J** [comment]
> Sorry for not having said it earlier. I'd have liked to take care of it myself, but it turns out I'm a bit busy, so if you would...

I can barely sprite (As you probably noticed), but I'll try my best

**Comment by YakumoChen** [comment]
Meme of the week pr

**Comment by Majkl-J** [comment]
Alright I updated it, I can give one more shot to the front part if that's still an issue but that's the most I can really manage

**Comment by ATH1909** [comment]
> That joke sucked, and of all the actually kinda good stuff in Orville, it makes me mad this is the one that took off.

I think the delivery on "five HUNDRED cigarettes" is what sells it (+ that YTP), less so the intended joke itself

**Pull request merged**